sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetEncodingSuite.scala

```diff
@@ -78,4 +78,29 @@ class ParquetEncodingSuite extends ParquetCompatibilityTest with SharedSQLContex
       }}
     }
   }
+
+  test("Read row group containing both dictionary and plain encoded pages") {
+    spark.conf.set("parquet.dictionary.page.size", "2048")
+    spark.conf.set("parquet.page.size", "4096")
+
+    withTempPath { dir =>
+      // In order to explicitly test for SPARK-14217, we set the parquet dictionary and page size
+      // such that the following data spans across 3 pages (within a single row group) where the
+      // first page is dictionary encoded and the remaining two are plain encoded.
+      val data = (0 until 512).flatMap(i => List(i.toString, i.toString, i.toString))
+      data.toDF("f").coalesce(1).write.parquet(dir.getCanonicalPath)
+      val file = SpecificParquetRecordReaderBase.listDirectory(dir).toArray.head
+
+      val reader = new VectorizedParquetRecordReader
+      reader.initialize(file.asInstanceOf[String], null)
+      val batch = reader.resultBatch()
+      assert(reader.nextBatch())
+
+      (0 until 512).foreach { i =>
+        assert(batch.column(0).getUTF8String(3 * i).toString == i.toString)
```

Comment:
Two things here:

1. Create column in line 96 (`batch.column(0)`).
2. Since you convert `toString`, what do you think about `toInt` instead (since `i` is `Int` anyway). One conversion less :)